File: 7.java
class Solution {
    public int reverse(int x) {
        String s="";
        for(int i=0;i<(""+Math.abs(x)).length();i++){
            s=(""+Math.abs(x)).charAt(i)+s;
        }
        try{
            if(x<0) return Integer.parseInt("-"+s);
            else return Integer.parseInt(s);
        }catch(Exception e){
            return 0;
        }
    }
}

File: 7.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x<0:
            return -self.reverse(-x)
        else:
            t = int(str(x%2**31)[::-1])
            if t > 2**31-1 or t < -(2**31):
                return 0
            else:
                return t
